internal/core/software_renderer.rs

```diff
@@ -791,6 +791,34 @@ impl RendererSealed for SoftwareRenderer {
         *self.maybe_window_adapter.borrow_mut() = Some(Rc::downgrade(window_adapter));
         self.partial_cache.borrow_mut().clear();
     }
+
+    fn screenshot(&self) -> Result<SharedImageBuffer, PlatformError> {
+        let Some(window_adapter) =
+            self.maybe_window_adapter.borrow().as_ref().and_then(|w| w.upgrade())
+        else {
+            return Err(
+                "SoftwareRenderer's screenshot called without a window adapter present".into()
+            );
+        };
+
+        let window = window_adapter.window();
+        let size = window.size();
+
+        let Some((width, height)) = size.width.try_into().ok().zip(size.height.try_into().ok())
+        else {
+            // Nothing to render
+            return Err("grab_window() called on window with invalid size".into());
+        };
+
+        let mut target_buffer = SharedPixelBuffer::<crate::graphics::Rgb8Pixel>::new(width, height);
+
+        self.set_repaint_buffer_type(RepaintBufferType::NewBuffer);
+        let _ = self.render(target_buffer.make_mut_slice(), width as usize);
```

Comment:
```suggestion
        self.render(target_buffer.make_mut_slice(), width as usize);
```

The function is not `must_use`  so the `let _` is misleading.
If one day we change render to return a Result, then we would want to have the warning or error.

Reply:
Ah great, yes, if that works :)